### PR TITLE
Limit retention period

### PR DESCRIPTION
**Pull request by midigofrank**
### Description

This PR adds an extension that allows for limiting retention period.

The extension allows for 3 things:

1. Only show desired retention periods in the project settings page
2. Show a message next to the select input if provided
3. Add custom validation to the `project changeset` when calling `Projects.update_project/2`

I also found a refactoring opportunity. I combined `ProjectAlertsLimiter` and `ProjectUsersLimiter` into `ProjectLimiter` to avoid adding extra modules.

### Validation steps

This can be validated in the billing app. There are also tests that validate the functionality

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [ ] Code generation (copilot but not intellisense)
- [ ] Learning or fact checking
- [ ] Strategy / design
- [ ] Optimisation / refactoring
- [ ] Translation / spellchecking / doc gen
- [ ] Other
- [x] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

### Pre-submission checklist

- [x] I have performed a **self-review** of my code.
- [x] I have implemented and tested all related **authorization policies**. (e.g., `:owner`, `:admin`, `:editor`, `:viewer`)
- [x] I have updated the **changelog**.
- [x] I have ticked a box in "AI usage" in this PR

### CODE REVIEW

File: lib/lightning/projects/project.ex [hunk 1]
```diff
@@ -67,8 +65,6 @@ defmodule Lightning.Projects.Project do
     |> validate_length(:description, max: 240)
     |> validate_required([:name])
     |> validate_format(:name, ~r/^[a-z\-\d]+$/)
-    |> validate_inclusion(:history_retention_period, @retention_periods)
-    |> validate_inclusion(:dataclip_retention_period, @retention_periods)
```
Comment (stuartc) [inline lib/lightning/projects/project.ex:71]: I feel this should still be validated no? We're extending the validation via a hook, but these constraints are still valid here right?

File: lib/lightning_web/live/project_live/invite_collaborator_component.ex [hunk 2]
```diff
@@ -91,7 +91,7 @@ defmodule LightningWeb.ProjectLive.InviteCollaboratorComponent do
 
     project_users = Ecto.Changeset.get_embed(changeset, :invited_collaborators)
 
-    case ProjectUsersLimiter.request_new(
+    case ProjectLimiter.request_new_user(
```
Comment (stuartc) [inline lib/lightning_web/live/project_live/invite_collaborator_component.ex:94]: 👌 nice, I'm liking the renaming and conforming of the ProjectLimiter changes.

File: lib/lightning_web/live/project_live/settings.ex [hunk 3]
```diff
@@ -28,6 +27,7 @@ defmodule LightningWeb.ProjectLive.Settings do
   on_mount {LightningWeb.Hooks, :project_scope}
   on_mount {LightningWeb.Hooks, :limit_github_sync}
   on_mount {LightningWeb.Hooks, :limit_mfa}
+  on_mount {LightningWeb.Hooks, :limit_retention_periods}
```
Comment (stuartc) [inline lib/lightning_web/live/project_live/settings.ex:30]: Thinking out loud here, I think these hooks are needing a rename, afaik all the hooks in the file are related to projects.

Also, at least 2/3 of these are only used in this live view, so we might be extending an 'implicit contract' - where this one live view expects a bunch of assigns that are added by another module.

I don't necessarily thing we need to change this now - but the settings module (when using hooks) feels a bit 'stringy' where we quietly plug in a bunch of assigns that are all very specific.

File: lib/lightning_web/live/project_live/settings.html.heex [hunk 4]
```diff
@@ -898,19 +898,37 @@
...
                 </div>
 
-                <div class="">
+                <div class="flex gap-4 items-center">
                   <.input
                     type="select"
                     prompt="Select Period"
                     options={
-                      Enum.map([7, 14, 30, 90, 180, 365], fn days ->
+                      Enum.map(@data_retention_periods, fn days ->
```
Comment (stuartc) [inline lib/lightning_web/live/project_live/settings.html.heex:906]: Oh thank goodness 😅

**Review by stuartc — COMMENTED** [review]
Very nice, good work! Just one thing about the moving of the validations on the retention periods, could you take a look at my comment?

File: lib/lightning_web/live/project_live/settings.ex (hunk 3 above)
Comment (midigofrank) [reply lib/lightning_web/live/project_live/settings.ex:30]: Yeah, good point. Feels like we need a central place to extend the settings live view with limiters and extensions

**Review by stuartc — APPROVED** [review]

**Pull request merged**